The extension macro designed to add `Sendable` conformance generates an unexpected warning

**Description**  
The extension macro designed to add `Sendable` conformance generates an unexpected warning.

**Steps to Reproduce**  
1. Using Xcode 15.0, implement a `Sendable` macro as shown in the following link: [Apple Swift GitHub Repository](https://github.com/apple/swift/blob/40362469cacce597944f38fdba8a245ff0335f46/test/Serialization/Inputs/def_macro_plugin.swift#L55C1-L78C2).

    ```swift
    public struct SendableMacro: ExtensionMacro {
      public static func expansion(
        of node: AttributeSyntax,
        attachedTo: some DeclGroupSyntax,
        providingExtensionsOf type: some TypeSyntaxProtocol,
        conformingTo protocols: [TypeSyntax],
        in context: some MacroExpansionContext
      ) throws -> [ExtensionDeclSyntax] {
        if protocols.isEmpty {
          return []
        }
    
        let sendableExtension: DeclSyntax =
          """
          extension \(type.trimmed): Sendable {}
          """
    
        guard let extensionDecl = sendableExtension.as(ExtensionDeclSyntax.self) else {
          return []
        }
    
        return [extensionDecl]
      }
    }
    
    @main
    struct AddSendablePlugin: CompilerPlugin {
        let providingMacros: [Macro.Type] = [
            SendableMacro.self
        ]
    }
    ```

2. Create an interface:

    ```swift
    @attached(extension, conformances: Sendable)
    public macro AddSendable() = #externalMacro(module: "AddSendableMacros", type: "SendableMacro")
    ```
  
3. Utilize the macro in an executable:

    ```swift
    import AddSendable
    
    @AddSendable
    final class Message {
    }
    ```
4. Build it using `swift build`.
5. Produced warning:
```
/var/folders/lj/1_d6kbt94zb2wb7dtqvnl4kr0000gn/T/swift-generated-sources/@__swiftmacro_17AddSendableClient7Message0aB0fMe_.swift:1:1: warning: conformance to 'Sendable' must occur in the same source file as class 'Message'; use '@unchecked Sendable' for retroactive conformance
extension Message: Sendable {
^
AddSendable/Sources/AddSendableClient/main.swift:4:7: note: in expansion of macro 'AddSendable' here
final class Message {
      ^~~~~~~~~~~~~~~
```

**Environment**
- Swift compiler version: swift-driver 1.87.1, Apple Swift 5.9 (swiftlang-5.9.0.128.108 clang-1500.0.40.1)
- Xcode version: 15.0 (Build version 15A240d)
- Deployment target: macOS 14.0 (23A339)

--- a/lib/Sema/TypeCheckConcurrency.cpp
+++ b/lib/Sema/TypeCheckConcurrency.cpp
@@ -5104,8 +5104,9 @@ bool swift::checkSendableConformance(
   auto conformanceDecl = conformanceDC->getAsDecl();
   auto behavior = SendableCheckContext(conformanceDC, check)
       .defaultDiagnosticBehavior();
-  if (conformanceDC->getParentSourceFile() &&
-      conformanceDC->getParentSourceFile() != nominal->getParentSourceFile()) {
+  if (conformanceDC->getOutermostParentSourceFile() &&
+      conformanceDC->getOutermostParentSourceFile() !=
+      nominal->getOutermostParentSourceFile()) {
     conformanceDecl->diagnose(diag::concurrent_value_outside_source_file,
                               nominal)
       .limitBehavior(behavior);

--- a/test/Macros/Inputs/syntax_macro_definitions.swift
+++ b/test/Macros/Inputs/syntax_macro_definitions.swift
@@ -2057,6 +2057,31 @@ extension RequiredDefaultInitMacro: MemberMacro {
   }
 }
 
+public struct SendableMacro: ExtensionMacro {
+  public static func expansion(
+    of node: AttributeSyntax,
+    attachedTo decl: some DeclGroupSyntax,
+    providingExtensionsOf type: some TypeSyntaxProtocol,
+    conformingTo protocols: [TypeSyntax],
+    in context: some MacroExpansionContext
+  ) throws -> [ExtensionDeclSyntax] {
+    if protocols.isEmpty {
+      return []
+    }
+
+    let decl: DeclSyntax =
+      """
+      extension \(type.trimmed): Sendable {
+      }
+
+      """
+
+    return [
+      decl.cast(ExtensionDeclSyntax.self)
+    ]
+  }
+}
+
 public struct FakeCodeItemMacro: DeclarationMacro, PeerMacro {
   public static func expansion(
     of node: some FreestandingMacroExpansionSyntax,

--- a/test/Macros/macro_expand_extensions.swift
+++ b/test/Macros/macro_expand_extensions.swift
@@ -244,3 +244,15 @@ func testStringFoo(s: String) {
   "Test".printFoo()
   s.printFoo()
 }
+
+@attached(extension, conformances: Sendable)
+macro AddSendable() = #externalMacro(module: "MacroDefinition", type: "SendableMacro")
+
+@AddSendable
+final class SendableClass {
+}
+
+// expected-warning@+2 {{non-final class 'InvalidSendableClass' cannot conform to 'Sendable'; use '@unchecked Sendable'}}
+@AddSendable
+class InvalidSendableClass {
+}

